File: Cores/DotNet/Cybernatics/IntervalValuedDynamicSystem.cs
using System;
using OSOL.Extremum.Cores.DotNet.Arithmetics;
using OSOL.Extremum.Cores.DotNet.Vectors;

namespace OSOL.Extremum.Cores.DotNet.Cybernatics
{
    public class IntervalValuedDynamicSystem : DynamicSystem<Interval>
    {
        public IntervalValuedDynamicSystem(Func<VectorObject<Interval>, VectorObject<Interval>> f, Func<VectorObject<Interval>, VectorObject<Interval>> u, ButcherTableau butcherTableau)
        {
            this.f = f;
            this.u = u;
            this.butcherTableau = butcherTableau;
        }

        public sealed override double BaseToDouble(Interval b) => b.MiddlePoint;

        public sealed override Interval DoubleToBase(double d) => d;
    }
}

File: Cores/JVM/src/main/scala/OSOL/Extremum/Cores/JVM/Cybernatics/IntervalValuedDynamicSystem.scala
package OSOL.Extremum.Cores.JVM.Cybernatics

import OSOL.Extremum.Cores.JVM.Arithmetics.Interval
import OSOL.Extremum.Cores.JVM.Vectors.VectorObject

class IntervalValuedDynamicSystem(f: VectorObject[Interval] => VectorObject[Interval], u: VectorObject[Interval] => VectorObject[Interval], butcherTableau: ButcherTableau)
  extends DynamicSystem[Interval](f, u, butcherTableau) {

  final override implicit def doubleToBase(d: Double): Interval = Interval(d)

  final override implicit def baseToDouble(b: Interval): Double = b.middlePoint

}
